src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -105,13 +105,19 @@ UniqueColumnCombinations StoredTableNode::unique_column_combinations() const {
       continue;
     }
 
+    // We may only use the key constraints as UCCs for optimization purposes if they are certainly still valid,
+    // otherwise these optimizations could produce invalid query results.
+    if (!table->constraint_guaranteed_to_be_valid(table_key_constraint)) {
+      continue;
+    }
```

Comment:
You discard a possibly valid UCC in an edge case here: Imagine UCC was valid for CommitID 5. Our transaction runs on a snapshot w/ CommitID 7. Another transaction commited with CommitID 8 and inserted tuples that could possibly invalidate the UCC. However, the UCC is still valid for the snapshot that out transcation sees. We cannot derive this knowledge only using the max begin CID, so we should add a comment here.